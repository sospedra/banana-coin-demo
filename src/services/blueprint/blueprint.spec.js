/* global expect, describe, it */
const { toPlainObject, noop } = require('lodash')
const { composeBlueprint } = require('./')

const routeWithBlueprint = (blueprint) => {
  const route = () => {}
  route.blueprint = blueprint
  return route
}

describe('service blueprint', () => {
  it('should compose a blueprint string from routes object', () => {
    const routes = toPlainObject([
      routeWithBlueprint('# Cowboy'),
      routeWithBlueprint('# Bebop'),
      noop
    ])

    expect(composeBlueprint(routes)).toMatchSnapshot()
  })
})
