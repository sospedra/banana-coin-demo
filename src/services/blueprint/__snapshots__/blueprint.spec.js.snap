// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`service blueprint should compose a blueprint string from routes object 1`] = `
Array [
  "# Cowboy",
  "# Bebop",
]
`;
